### DISCUSSION TITLE
CouchDB to MSSQL

### DISCUSSION THREAD

**Discussion by kasgithub8** [Budibase/budibase#7470] [Help] [answered]
Hey,

is there a way to export the data from CouchDB and import them into a MsSQL database?

**Comment by melohagan** [comment]
Hey @kasgithub8 

I'm wondering why you would want to do this? Are you planning to migrate away from the Budibase Internal DB to SQL Server?

**Reply by kasgithub8** [reply]
@melohagan yeah so, I would love to do reports with that data.

**Comment by mike12345567** [comment, marked as answer]
Hi @kasgithub8 - the raw export from CouchDB to MS-SQL isn't too hard, you can export CSVs/JSON from Budibase tables and import these into SQL tables, however this doesn't cover a lot of the higher functionality that Budibase DB provides (formulas, relationships, attachments, etc). There isn't an easy way to do this manually right now - we've been researching this feature but its very complex.

We will hopefully implement something towards this in the future, but for now its a manual process of building SQL queries to bulk write your data from CouchDB into an SQL database. If you are only using native data types like strings, numbers, booleans etc then you can export a CSV and import from this as a bulk query statement, but otherwise it is not a simple process.

**Reply by kasgithub8** [reply]
Damn.
But it's definitely good to hear that you guys are looking into it!
Thanks!
